aashish/service.tf

```diff
@@ -0,0 +1,69 @@
+output "service_private_key" {
+  value     = tls_private_key.service.private_key_pem
+  sensitive = true
+}
+
+variable "service" {
+  type = list(object({
+    name          = string
+    ami           = string,
+    instance_type = string,
+    az            = number
+  }))
+}
+
+resource "tls_private_key" "service" {
+  algorithm = "RSA"
+}
+
+resource "aws_key_pair" "service" {
+  key_name   = "service"
+  public_key = tls_private_key.service.public_key_openssh
+  tags       = local.tags
+}
+
+resource "aws_security_group" "service" {
+  name   = "${local.project_name}-service-sg"
+  vpc_id = aws_vpc.main_vpc.id
+
+  ingress = [{
+    cidr_blocks      = []
+    description      = "Local"
+    from_port        = 0
+    protocol         = -1
+    to_port          = 0
+    ipv6_cidr_blocks = []
+    prefix_list_ids  = []
+    security_groups  = [aws_security_group.bastion.id]
+    self             = false
+  }]
+
+  egress = [{
+    cidr_blocks      = [local.default_route]
+    description      = "Public"
+    from_port        = 0
+    protocol         = -1
+    to_port          = 0
+    ipv6_cidr_blocks = []
+    prefix_list_ids  = []
+    security_groups  = []
+    self             = false
+  }]
+
+  tags = merge({
+    Name = "${local.project_name}-service-sg"
+  }, local.tags)
+}
+
+resource "aws_instance" "service" {
+  count                  = length(var.service)
```

Comment:
The count has a challenge, if you reduce it, you don't know which instance it will delete. Try using hasmaps.

Reply:
Converted to hashmap and used foreach instead